### ISSUE TITLE
AWX Backup Role Not using postgresConfigurationSecret 

### ISSUE THREAD

**Issue by r1200gsdave** [ansible/tower-operator#1577]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

I have an AWX deplyment in AWX EKS with an RDS postgresql database.  Config for DB is in the standard secret.  I am trying to use the AWX Backup role but the pod just gets through to the postgres.yml playbook and fails on the first step.

It is my understanding that it should find the secret for the database and use it but that does not seem to be happening.  I have tried with and without adding the secret name to the spec for the deployment and it still fails on Get PostgreSQL configuration step. 

fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: list object has no element 0\n\nThe error appears to be in '/opt/ansible/roles/backup/tasks/postgres.yml': line 3, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Get PostgreSQL configuration\n  ^ here\n"}

### AWX Operator version

2.6.0

### AWX version

23.2.0

### Kubernetes platform

other (please specify in additional information)

### Kubernetes/Platform version

EKS

### Modifications

no

### Steps to reproduce

create pod

### Expected results

pod stays runnign with out error

### Actual results

pod starts and then errors out and terminates

**Comment by r1200gsdave** [comment]
I believe that I have found something that might help this issue.

In the Backup Role postgres.yml file the play uses k8s_info to get to the secret which I beliee is erroring out as an unset variable in my install.

In the Restore Role the postgres secrets are gathered in a different manner, there a fact set which is either from a spec variable ot the deploy variable.

The restore role seems to be a more reliable method.

**Comment by fosterseth** [comment]
@rooftopcellist @TheRealHaoLiu is this a bug?

**Comment by TheRealHaoLiu** [comment]
https://github.com/ansible/awx-operator/blob/devel/roles/backup/tasks/postgres.yml/#L3-L25

from this playbook it looks like we are getting the postgres secret from the status of AWX resource and than using the information from the secret itself to perform the backup

can u do a `kubectl get awx -o yaml` and `kubectl get awxbackup -o yaml` and paste the information in this issue

@rooftopcellist, @r1200gsdave is right it does seem like the restore role's method of getting postgres config secret is much more reliable

**Comment by r1200gsdave** [comment]
kubectl get awx -o yaml -n awx
```yaml
apiVersion: v1
items:
- apiVersion: awx.ansible.com/v1beta1
  kind: AWX
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"awx.ansible.com/v1beta1","kind":"AWX","metadata":{"annotations":{},"name":"awx","namespace":"awx"},"spec":{"hostname":"awx-test.exactsciences.net","ingress_annotations":"alb.ingress.kubernetes.io/listen-ports: '[{\"HTTPS\":443}]'\nalb.ingress.kubernetes.io/actions.redirect: \"{\\\"Type\\\": \\\"redirect\\\", \\\"RedirectConfig\\\": {\\\"Protocol\\\": \\\"HTTPS\\\", \\\"Port\\\": \\\"443\\\", \\\"StatusCode\\\": \\\"HTTP_301\\\"}}\"\nalb.ingress.kubernetes.io/scheme: internal\nalb.ingress.kubernetes.io/target-type: ip\nkubernetes.io/ingress.class: alb\nalb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-east-1:767914017133:certificate/1539bfcb-215d-47cd-bbdc-f1eebbeb0a8e\nalb.ingress.kubernetes.io/load-balancer-attributes: \"idle_timeout.timeout_seconds=360\"\n","ingress_path":"/","ingress_path_type":"Prefix","ingress_type":"ingress","postgres_configuration_secret":"awx-postgres-configuration","service_type":"nodeport"}}
    creationTimestamp: "2023-09-28T19:26:56Z"
    generation: 1
    labels:
      app.kubernetes.io/component: awx
      app.kubernetes.io/managed-by: awx-operator
      app.kubernetes.io/operator-version: 2.6.0
      app.kubernetes.io/part-of: awx
    name: awx
    namespace: awx
    resourceVersion: "24630808"
    uid: d52d33f5-d674-444e-909c-71ba4e140f35
  spec:
    admin_user: admin
    auto_upgrade: true
    create_preload_data: true
    garbage_collect_secrets: false
    hostname: awx-test.exactsciences.net
    image_pull_policy: IfNotPresent
    ingress_annotations: |
      alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS":443}]'
      alb.ingress.kubernetes.io/actions.redirect: "{\"Type\": \"redirect\", \"RedirectConfig\": {\"Protocol\": \"HTTPS\", \"Port\": \"443\", \"StatusCode\": \"HTTP_301\"}}"
      alb.ingress.kubernetes.io/scheme: internal
      alb.ingress.kubernetes.io/target-type: ip
      kubernetes.io/ingress.class: alb
      alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-east-1:767914017133:certificate/1539bfcb-215d-47cd-bbdc-f1eebbeb0a8e
      alb.ingress.kubernetes.io/load-balancer-attributes: "idle_timeout.timeout_seconds=360"
    ingress_path: /
    ingress_path_type: Prefix
    ingress_type: ingress
    ipv6_disabled: false
    loadbalancer_ip: ""
    loadbalancer_port: 80
    loadbalancer_protocol: http
    no_log: true
    postgres_configuration_secret: awx-postgres-configuration
    postgres_keepalives: true
    postgres_keepalives_count: 5
    postgres_keepalives_idle: 5
    postgres_keepalives_interval: 5
    projects_persistence: false
    projects_storage_access_mode: ReadWriteMany
    projects_storage_size: 8Gi
    replicas: 1
    route_tls_termination_mechanism: Edge
    service_type: nodeport
    set_self_labels: true
    task_privileged: false
  status:
    adminPasswordSecret: awx-admin-password
    adminUser: admin
    broadcastWebsocketSecret: awx-broadcast-websocket
    conditions:
    - lastTransitionTime: "2023-09-28T19:32:16Z"
      reason: ""
      status: "False"
      type: Failure
    - lastTransitionTime: "2023-09-28T19:27:08Z"
      reason: Successful
      status: "True"
      type: Running
    - lastTransitionTime: "2023-09-29T16:31:01Z"
      reason: Successful
      status: "True"
      type: Successful
    image: quay.io/ansible/awx:23.2.0
    postgresConfigurationSecret: awx-postgres-configuration
    secretKeySecret: awx-secret-key
    version: 23.2.0
kind: List
metadata:
  resourceVersion: ""
```
ubuntu@ip-172-29-228-228:~$ kubectl apply -f awxbackup.yml
awxbackup.awx.ansible.com/awxbackup-2023-09-28 created



kubectl get awxbackup -o yaml -n awx
```yaml
apiVersion: v1
items:
- apiVersion: awx.ansible.com/v1beta1
  kind: AWXBackup
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"awx.ansible.com/v1beta1","kind":"AWXBackup","metadata":{"annotations":{},"name":"awxbackup-2023-09-28","namespace":"awx"},"spec":{"deployment_name":"awx","no_log":false,"postgres_configuration_secret":"awx-postgres-configuration"}}
    creationTimestamp: "2023-10-04T20:58:25Z"
    finalizers:
    - awx.ansible.com/finalizer
    generation: 1
    labels:
      app.kubernetes.io/component: awx
      app.kubernetes.io/managed-by: awx-operator
      app.kubernetes.io/operator-version: 2.6.0
      app.kubernetes.io/part-of: awxbackup-2023-09-28
    name: awxbackup-2023-09-28
    namespace: awx
    resourceVersion: "26987079"
    uid: cb958bfa-8ed9-45fe-a227-15996ddddb2a
  spec:
    deployment_name: awx
    image_pull_policy: IfNotPresent
    no_log: false
    postgres_configuration_secret: awx-postgres-configuration
    set_self_labels: true
  status:
    conditions:
    - lastTransitionTime: "2023-10-04T20:58:25Z"
      reason: Running
      status: "True"
      type: Running
kind: List
metadata:
  resourceVersion: ""
```

**Comment by r1200gsdave** [comment]
this is the error
****\r\ntask path: /opt/ansible/roles/backup/tasks/postgres.yml:3\nfatal: [localhost]: FAILED! => {\"msg\": \"The task includes an option with an undefined variable. The error was: list object has no element 0\\n\\nThe error appears to be in '/opt/ansible/roles/backup/tasks/postgres.yml': line 3, column 3, but may\\nbe elsewhere in the file depending on the exact syntax problem.\\n\\nThe offending line appears to be:\\n\\n\\n- name: Get PostgreSQL configuration\\n  ^ here\\n\"}\n\r\

I believe that in line 7 of the postgres.yaml file the element 0 does not exist

  name: "{{ this_awx['resources'][0]['status']['postgresConfigurationSecret'] }}"

**Comment by dsavineau** [comment]
It would be great to get the logs (ideally with verbosity increased) from prior tasks especially the one that registers the `this_awx` variable [1]

From the output shared above, we can see that the awx object has `postgresConfigurationSecret` correctly configured in the status.

[1] https://github.com/ansible/awx-operator/blob/devel/roles/backup/tasks/init.yml#L88-L94

**Comment by romankuzmik** [comment]
- backup/tasks/init.yml reads data from kind: "AWX", not from "AWXBackup"
- backup/tasks/postgres.yml.yml expects kind: "AWX", to have status dict with postgresConfigurationSecret attribite set

but, kind: "AWX", has nothing in status at all:

```
kubectl get awx -o yaml -n awx
...
  status:
    conditions:
    - lastTransitionTime: "2023-10-05T16:27:50Z"
      reason: ""
      status: "False"
      type: Failure
    - lastTransitionTime: "2023-10-05T16:27:50Z"
      reason: Successful
      status: "True"
      type: Running
    - lastTransitionTime: "2023-10-05T16:28:22Z"
      reason: Successful
      status: "True"
      type: Successful
```

**Comment by dsavineau** [comment]
> but, kind: "AWX", has nothing in status at all:

It should and from the log shared in https://github.com/ansible/awx-operator/issues/1577#issuecomment-1747632901 you can see it.
I'll paste it below (but simplified)

```yaml
apiVersion: v1
items:
- apiVersion: awx.ansible.com/v1beta1
  kind: AWX
  metadata:
    (...)
    name: awx
    namespace: awx
  spec:
    (...)
  status:
    (...)
    postgresConfigurationSecret: awx-postgres-configuration  <--- HERE
```

This status should be set during the initial deployment so if it's not there then there's something wrong with that deployment.

https://github.com/ansible/awx-operator/blob/devel/roles/installer/tasks/update_status.yml#L20-L27

**Comment by r1200gsdave** [comment]
I can provide logs if you tell me how to get to the verbosity you need.

**Comment by dsavineau** [comment]
You can add `"ansible.sdk.operatorframework.io/verbosity": "5"` to your AWXBackup CR metadata/annotations

https://sdk.operatorframework.io/docs/building-operators/ansible/reference/advanced_options/#ansible-verbosity

**Comment by dsavineau** [comment]
you can also set `no_log: false` in the AWXBackup CR spec to avoid some tasks output to be hidden

**Comment by r1200gsdave** [comment]
I have added that verbosity and it did not add log entries

**Comment by dsavineau** [comment]
```yaml
apiVersion: awx.ansible.com/v1beta1
kind: AWXBackup
metadata:
  annotations:
    ansible.sdk.operatorframework.io/verbosity: '5'
  name: awxbackup
  namespace: awx
spec:
  deployment_name: awx
  no_log: false
```
I just tried this and the backup was successful with verbosity increased in the logs
Have you just updated the existing `awxbackup-2023-09-28` CR or delete + re-create ?

```console
TASK [backup : Look up details for this deployment] ****************************
task path: /opt/ansible/roles/backup/tasks/init.yml:88

<localhost> ESTABLISH LOCAL CONNECTION FOR USER: 1000660000
<localhost> EXEC /bin/sh -c 'echo ~1000660000 && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /opt/ansible/.ansible/tmp `"&& mkdir "` echo /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958 `" && echo ansible-tmp-1696538296.1426015-2758-37093776501958="` echo /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958 `" ) && sleep 0'_info.py
<localhost> PUT /opt/ansible/.ansible/tmp/ansible-local-2620d0yahool/tmp77iwc5o9 TO /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958/AnsiballZ_k8s_info.py
<localhost> EXEC /bin/sh -c 'chmod u+x /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958/ /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958/AnsiballZ_k8s_info.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python3.8 /opt/ansible/.ansible/tmp/ansible-tmp-1696538296.1426015-2758-37093776501958/AnsiballZ_k8s_info.py && sleep 0'
{"level":"info","ts":"2023-10-05T20:38:16Z","logger":"proxy","msg":"Read object from cache","resource":{"IsResourceRequest":true,"Path":"/apis/awx.ansible.com/v1beta1/namespaces/awx/awxs/awx","Verb":"get","APIPrefix":"apis","APIGroup":"awx.ansible.com","APIVersion":"v1beta1","Namespace":"awx","Resource":"awxs","Subresource":"","Name":"awx","Parts":["awxs","awx"]}}2>&1 && sleep 0'
ok: [localhost] => {
    "api_found": true,
    "changed": false,
(...)
    "resources": [
        {
            "apiVersion": "awx.ansible.com/v1beta1",
            "kind": "AWX",
(...)
            "status": {
                "URL": "https://awx-awx.apps-crc.testing",
                "adminPasswordSecret": "awx-admin-password",
                "adminUser": "admin",
                "broadcastWebsocketSecret": "awx-broadcast-websocket",
                "conditions": [
                    {
                        "lastTransitionTime": "2023-10-05T20:34:43Z",
                        "reason": "",
                        "status": "False",
                        "type": "Failure"
                    },
                    {
                        "lastTransitionTime": "2023-10-05T20:27:53Z",
                        "reason": "Successful",
                        "status": "True",
                        "type": "Running"
                        "lastTransitionTime": "2023-10-05T20:36:00Z",
                        "reason": "Successful",
                        "status": "True",
                        "type": "Successful"
                    }
                ],
                "image": "quay.io/ansible/awx:23.1.0",
                "postgresConfigurationSecret": "awx-postgres-configuration",
                "secretKeySecret": "awx-secret-key",
                "version": "23.1.0"
            }
        }
    ]
}
```

**Comment by r1200gsdave** [comment]
TASK [backup : Look up details for this deployment] ****************************
task path: /opt/ansible/roles/backup/tasks/init.yml:88

<localhost> ESTABLISH LOCAL CONNECTION FOR USER: ansible
<localhost> EXEC /bin/sh -c 'echo ~ansible && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /opt/ansible/.ansible/tmp `"&& mkdir "` echo /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564 `" && echo ansible-tmp-1696598559.8601475-3130-50655890299564="` echo /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564 `" ) && sleep 0'
Using module file /opt/ansible/.ansible/collections/ansible_collections/kubernetes/core/plugins/modules/k8s_info.py
<localhost> PUT /opt/ansible/.ansible/tmp/ansible-local-29921k57ealr/tmpt9tjfkgm TO /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564/AnsiballZ_k8s_info.py
<localhost> EXEC /bin/sh -c 'chmod u+x /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564/ /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564/AnsiballZ_k8s_info.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python3.8 /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564/AnsiballZ_k8s_info.py && sleep 0'
{"level":"info","ts":"2023-10-06T13:22:40Z","logger":"proxy","msg":"Read object from cache","resource":{"IsResourceRequest":true,"Path":"/apis/awx.ansible.com/v1beta1/namespaces/awx/awxs/awx","Verb":"get","APIPrefix":"apis","APIGroup":"awx.ansible.com","APIVersion":"v1beta1","Namespace":"awx","Resource":"awxs","Subresource":"","Name":"awx","Parts":["awxs","awx"]}}
<localhost> EXEC /bin/sh -c 'rm -f -r /opt/ansible/.ansible/tmp/ansible-tmp-1696598559.8601475-3130-50655890299564/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {
    "api_found": true,
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": null,
            "api_version": "awx.ansible.com/v1beta1",
            "ca_cert": null,
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": null,
            "impersonate_groups": null,
            "impersonate_user": null,
            "kind": "AWX",
            "kubeconfig": null,
            "label_selectors": [],
            "name": "awx",
            "namespace": "awx",
            "no_proxy": null,
            "password": null,
            "persist_config": null,
            "proxy": null,
            "proxy_headers": null,
            "username": null,
            "validate_certs": null,
            "wait": false,
            "wait_condition": null,
            "wait_sleep": 5,
            "wait_timeout": 120
        }
    },
    "resources": [
        {
            "apiVersion": "awx.ansible.com/v1beta1",
            "kind": "AWX",
            "metadata": {
                "annotations": {
                    "kubectl.kubernetes.io/last-applied-configuration": "{\"apiVersion\":\"awx.ansible.com/v1beta1\",\"kind\":\"AWX\",\"metadata\":{\"annotations\":{},\"name\":\"awx\",\"namespace\":\"awx\"},\"spec\":{\"hostname\":\"awx-test.exactsciences.net\",\"ingress_annotations\":\"alb.ingress.kubernetes.io/listen-ports: '[{\\\"HTTPS\\\":443}]'\\nalb.ingress.kubernetes.io/actions.redirect: \\\"{\\\\\\\"Type\\\\\\\": \\\\\\\"redirect\\\\\\\", \\\\\\\"RedirectConfig\\\\\\\": {\\\\\\\"Protocol\\\\\\\": \\\\\\\"HTTPS\\\\\\\", \\\\\\\"Port\\\\\\\": \\\\\\\"443\\\\\\\", \\\\\\\"StatusCode\\\\\\\": \\\\\\\"HTTP_301\\\\\\\"}}\\\"\\nalb.ingress.kubernetes.io/scheme: internal\\nalb.ingress.kubernetes.io/target-type: ip\\nkubernetes.io/ingress.class: alb\\nalb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-east-1:767914017133:certificate/1539bfcb-215d-47cd-bbdc-f1eebbeb0a8e\\nalb.ingress.kubernetes.io/load-balancer-attributes: \\\"idle_timeout.timeout_seconds=360\\\"\\n\",\"ingress_path\":\"/\",\"ingress_path_type\":\"Prefix\",\"ingress_type\":\"ingress\",\"postgres_configuration_secret\":\"awx-postgres-configuration\",\"service_type\":\"nodeport\"}}\n"
                },
                "creationTimestamp": "2023-09-28T19:26:56Z",
                "generation": 1,
                "labels": {
                    "app.kubernetes.io/component": "awx",
                    "app.kubernetes.io/managed-by": "awx-operator",
                    "app.kubernetes.io/operator-version": "2.6.0",
                    "app.kubernetes.io/part-of": "awx"
                },
                "managedFields": [
                    {
                        "apiVersion": "awx.ansible.com/v1beta1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:metadata": {
                                "f:annotations": {
                                    ".": {},
                                    "f:kubectl.kubernetes.io/last-applied-configuration": {}
                                }
                            },
                            "f:spec": {
                                ".": {},
                                "f:admin_user": {},
                                "f:auto_upgrade": {},
                                "f:create_preload_data": {},
                                "f:garbage_collect_secrets": {},
                                "f:hostname": {},
                                "f:image_pull_policy": {},
                                "f:ingress_annotations": {},
                                "f:ingress_path": {},
                                "f:ingress_path_type": {},
                                "f:ingress_type": {},
                                "f:ipv6_disabled": {},
                                "f:loadbalancer_ip": {},
                                "f:loadbalancer_port": {},
                                "f:loadbalancer_protocol": {},
                                "f:no_log": {},
                                "f:postgres_configuration_secret": {},
                                "f:projects_persistence": {},
                                "f:projects_storage_access_mode": {},
                                "f:projects_storage_size": {},
                                "f:replicas": {},
                                "f:route_tls_termination_mechanism": {},
                                "f:service_type": {},
                                "f:set_self_labels": {},
                                "f:task_privileged": {}
                            }
                        },
                        "manager": "kubectl-client-side-apply",
                        "operation": "Update",
                        "time": "2023-09-28T19:26:56Z"
                    },
                    {
                        "apiVersion": "awx.ansible.com/v1beta1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:metadata": {
                                "f:labels": {
                                    ".": {},
                                    "f:app.kubernetes.io/component": {},
                                    "f:app.kubernetes.io/managed-by": {},
                                    "f:app.kubernetes.io/operator-version": {},
                                    "f:app.kubernetes.io/part-of": {}
                                }
                            }
                        },
                        "manager": "OpenAPI-Generator",
                        "operation": "Update",
                        "time": "2023-09-29T16:27:25Z"
                    },
                    {
                        "apiVersion": "awx.ansible.com/v1beta1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:status": {
                                "f:adminPasswordSecret": {},
                                "f:adminUser": {},
                                "f:broadcastWebsocketSecret": {},
                                "f:image": {},
                                "f:postgresConfigurationSecret": {},
                                "f:secretKeySecret": {},
                                "f:version": {}
                            }
                        },
                        "manager": "OpenAPI-Generator",
                        "operation": "Update",
                        "subresource": "status",
                        "time": "2023-09-29T16:29:45Z"
                    },
                    {
                        "apiVersion": "awx.ansible.com/v1beta1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:status": {
                                ".": {},
                                "f:conditions": {}
                            }
                        },
                        "manager": "ansible-operator",
                        "operation": "Update",
                        "subresource": "status",
                        "time": "2023-09-29T16:31:01Z"
                    }
                ],
                "name": "awx",
                "namespace": "awx",
                "resourceVersion": "24630808",
                "uid": "d52d33f5-d674-444e-909c-71ba4e140f35"
            },
            "spec": {
                "admin_user": "admin",
                "auto_upgrade": true,
                "create_preload_data": true,
                "garbage_collect_secrets": false,
                "hostname": "awx-test.exactsciences.net",
                "image_pull_policy": "IfNotPresent",
                "ingress_annotations": "alb.ingress.kubernetes.io/listen-ports: '[{\"HTTPS\":443}]'\nalb.ingress.kubernetes.io/actions.redirect: \"{\\\"Type\\\": \\\"redirect\\\", \\\"RedirectConfig\\\": {\\\"Protocol\\\": \\\"HTTPS\\\", \\\"Port\\\": \\\"443\\\", \\\"StatusCode\\\": \\\"HTTP_301\\\"}}\"\nalb.ingress.kubernetes.io/scheme: internal\nalb.ingress.kubernetes.io/target-type: ip\nkubernetes.io/ingress.class: alb\nalb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-east-1:767914017133:certificate/1539bfcb-215d-47cd-bbdc-f1eebbeb0a8e\nalb.ingress.kubernetes.io/load-balancer-attributes: \"idle_timeout.timeout_seconds=360\"\n",
                "ingress_path": "/",
                "ingress_path_type": "Prefix",
                "ingress_type": "ingress",
                "ipv6_disabled": false,
                "loadbalancer_ip": "",
                "loadbalancer_port": 80,
                "loadbalancer_protocol": "http",
                "no_log": true,
                "postgres_configuration_secret": "awx-postgres-configuration",
                "postgres_keepalives": true,
                "postgres_keepalives_count": 5,
                "postgres_keepalives_idle": 5,
                "postgres_keepalives_interval": 5,
                "projects_persistence": false,
                "projects_storage_access_mode": "ReadWriteMany",
                "projects_storage_size": "8Gi",
                "replicas": 1,
                "route_tls_termination_mechanism": "Edge",
                "service_type": "nodeport",
                "set_self_labels": true,
                "task_privileged": false
            },
            "status": {
                "adminPasswordSecret": "awx-admin-password",
                "adminUser": "admin",
                "broadcastWebsocketSecret": "awx-broadcast-websocket",
                "conditions": [
                    {
                        "lastTransitionTime": "2023-09-28T19:32:16Z",
                        "reason": "",
                        "status": "False",
                        "type": "Failure"
                    },
                    {
                        "lastTransitionTime": "2023-09-28T19:27:08Z",
                        "reason": "Successful",
                        "status": "True",
                        "type": "Running"
                    },
                    {
                        "lastTransitionTime": "2023-09-29T16:31:01Z",
                        "reason": "Successful",
                        "status": "True",
                        "type": "Successful"
                    }
                ],
                "image": "quay.io/ansible/awx:23.2.0",
                "postgresConfigurationSecret": "awx-postgres-configuration",
                "secretKeySecret": "awx-secret-key",
                "version": "23.2.0"
            }
        }
    ]
}

**Comment by r1200gsdave** [comment]
[awx-backup-log-10-06.txt](https://github.com/ansible/awx-operator/files/12831664/awx-backup-log-10-06.txt)
The cut and paste got clunky.... attaching a whole log

**Comment by kurokobo** [comment]
From your log, you are facing another issue: https://github.com/ansible/awx-operator/issues/1572

> fatal: [localhost]: FAILED! => {
>     "msg": "The conditional check ''Created partitions for' in result.stdout' failed. The error was: error while evaluating conditional ('Created partitions for' in result.stdout): 'dict object' has no attribute 'stdout'"
> }

Could you please name your AWXBackup as just `awx`, or use Operator 2.5.3?

**Comment by r1200gsdave** [comment]
I did rename the AWXBackup to just awx and recreated the CR.

A pod spun up and ran and then terminated.  There is no backup pod running. 

I see the PVC getting attached to the operator & task pods now.

 I ran the example playbook and got this error:

ERROR! the role 'backup' was not found in /runner/project/roles:/runner/requirements_roles:/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/runner/project

**Comment by dsavineau** [comment]
@r1200gsdave According to your logs, the `Look up details for this deployment` task always succeed and the `AWXBackup` CR is always using the right `postgresConfigurationSecret` value.
The error is related to something else (`Precreate database partitions` task)

```console
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check ''Created partitions for' in result.stdout' failed. The error was: error while evaluating conditional ('Created partitions for' in result.stdout): 'dict object' has no attribute 'stdout'"
}
```
This has been addressed in #1572 and #1573 so either stick to the 2.5.3 operator or wait for the 2.6.1 release.

**Comment by rooftopcellist** [comment]
It looks like this issue has been resolved, @r1200gsdave please open a new issue if that is not the case.  

Thanks!
